### PR TITLE
THREESCALE-10924: Remove duplicated config .yml files, keep only those in config/

### PR DESCRIPTION
**Pull request by mayorova**
**What this PR does / why we need it**:

We currently have multiple places where `.yml` configuration files are set:

- `config` - the default folder
- `config/examples` - these are not added to the container image, but for development they need to be copied to `config`
- `openshift/system/config` - these are copied to the container image by Dockerfile commands.

This causes some inconveniences:
- whenever the config changes, we need to update all the files
- we sometimes need to remember to copy the updated files from `example` to local versions again

This can be avoided by having a single set of configs. Normally, they would not collide, because we use `development` and `test` for development, and `production` typically is only used in the container image. And even if we wanted to run `production` locally, the values are normally configured through env variables, so it's the matter of adjusting your `.env` file.

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-10924

**Verification steps** 

See these comments:
https://github.com/3scale/porta/pull/3766#issuecomment-2079554409
https://github.com/3scale/porta/pull/3766#issuecomment-2082590997

**Special notes for your reviewer**:

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -253,7 +253,7 @@ def cache_store_config
     email_sanitizer_configs = (three_scale.delete(:email_sanitizer) || {}).symbolize_keys
     config.three_scale.email_sanitizer.merge!(email_sanitizer_configs)
 
-    config.three_scale.merge!(three_scale.slice!(:force_ssl, :access_code))
+    config.three_scale.merge!(three_scale.slice!(:force_ssl))
```
Comment (mayorova) [inline config/application.rb:256]: I think there was actually a bug here. This `access_code` is retrieved here: https://github.com/3scale/porta/blob/3f78b0f6d496c47513f51d5a9722cae9134070e3/lib/developer_portal/lib/access_code_protection.rb#L42

But with this code, it was resulting in `nil`. The correct way to get the value would be `Rails.application.config.access_code`. But I thought it would be better to place it under `three_scale` specific config.

File: openshift/system/config/prometheus.yml [hunk 2]
```diff
@@ -1,3 +0,0 @@
-production:
```
Comment (mayorova) [inline openshift/system/config/prometheus.yml:1]: This files is not used. It was initially added here: https://github.com/3scale/porta/commit/809e5dedfda87c1ae01ea37d5e91c58d8bff2b0a#diff-c1fd91cb1911a0512578b99f657554526f3e1421decdb9e908712beab57e10f9 , but then its loading was removed in https://github.com/3scale/porta/commit/79d15baae271c7e568ed351d9c41f5f6263b0324

**Comment by mayorova** [comment]
THIS COMMENT IS WIP: 

Differences between the current and the target configs, caused by the refactoring of the config files: 
`development`:

- `three_scale.access_code` added (was not existing before, due to a bug, explained [here](https://github.com/3scale/porta/pull/3766#discussion_r1579560580)
- `three_scale.core.fake_server: false` added (was not present before, the behavior is the same)
- `three_scale.currencies` - added a hash with a list of values, was empty hash before (2 default currencies were used in this case) - I think there is no reason to NOT have the full currencies list in development
- `three_scale.noreply_email` - changed from `"no-reply@example.net"` to `"no-reply@example.com"`
- `three_scale.notification_email` - changed from `"3scale Notification <no-reply@example.com>"` to `"example.com Notification <no-reply@example.com>"`
- `three_scale.segment.write_key` - changed from `"null"` to `true` - we don't really use it in development, and as it is disabled anyway, the value doesn't matter.
- `three_scale.service_discovery` - was full config, now is just `enabled: false`
- `smtp_settings` - was missing, now is full config

File: config/web_hooks.yml [hunk 3]
```diff
@@ -1,15 +1,14 @@
 base: &default
-  perform_deliveries: <%= ENV.fetch('WEBHOOK_PERFORM_DELIVERIES', '0') == '1' %>
+  perform_deliveries: <%= ENV.fetch('WEBHOOK_PERFORM_DELIVERIES', '1') == '1' %>
```
Comment (mayorova) [inline config/web_hooks.yml:2]: This is from https://github.com/3scale/porta/blob/539780953ebc6eddebcc1450a842e9d3198ea571/openshift/system/config/web_hooks.yml#L2

**Comment by mayorova** [comment]
Script that I used to print the configs:
```#!/usr/bin/ruby

config = {}

%i(web_hooks three_scale zync domain_substitution paperclip_defaults backend_client redis s3).each do |key|
  config[key] = Rails.configuration.send(key).sort.to_h
end

config[:cache_store] = Rails.configuration.cache_store

config[:smtp_settings] = Rails.configuration.action_mailer.smtp_settings

config[:zync] = {}
%i(endpoint authentication connect_timeout send_timeout receive_timeout skip_non_oidc_applications).each do |key|
  config[:zync][key] = Rails.configuration.zync.send(key)
end

config[:database] = ActiveRecord::Base.configurations[Rails.env]

pp config; nil
```

Example commands:

```
bundle exec rails runner /path/to/script.rb > master.txt
RAILS_ENV=production bundle exec rails runner /path/to/script.rb > prodmaster.txt
```

File: config/core.yml [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
-  # fake_server: 'http://localhost:3000/internal/'
-#  fake_server: false
+  fake_server: false
   username: <%= ENV.fetch('CONFIG_INTERNAL_API_USER', nil) %>
   password: <%= ENV.fetch('CONFIG_INTERNAL_API_PASSWORD', nil) %>
 
 development:
   <<: *default
+  # fake server is useful when you don't run a real backend apisonator
+  # fake_server: 'http://localhost:3000/internal/'
```
Comment (mayorova) [inline config/core.yml:10]: OK, this might be a potential place that a developer might want to change locally. We can add an env var for this.

File: config/database.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+base: &default
+<% case ENV['DATABASE_URL'].to_s
```
Comment (mayorova) [inline config/database.yml:2]: So, this might be controversial... 
This changed file does the `ENV['DATABASE_URL'].to_s` 4 times :grimacing:  (well, I guess at the execution time it will be 2.
But to be honest, for me it's much easier to understand this one than the original one: https://github.com/3scale/porta/blob/539780953ebc6eddebcc1450a842e9d3198ea571/config/examples/database.yml

But maybe I'm wrong, and the original is actually better. Opinions welcome!

Comment (mayorova) [reply config/database.yml:2]: I changed my mind :upside_down_face: 

I will revert the format of this file to the original one.

File: features/old/cms/email_templates.feature [hunk 6]
```diff
@@ -61,7 +61,7 @@ Feature: Email templates management
      And I follow "Sign up notification for provider"
      And I fill the form with following:
       | Bcc          | From |
-      | test@bcc.com | "Some Really Long String" <api@example.net> |
+      | test@bcc.com | "Some Really Long String" <api@example.com> |
```
[the remark this answers is not in the document]
Comment (mayorova) [reply features/old/cms/email_templates.feature:64]: Explanation in this commit's description: https://github.com/3scale/porta/pull/3766/commits/f036401e719c5e08a22adc28ea43e1c8bfd06658

File: config/segment.yml [hunk 7]
```diff
@@ -1,7 +1,7 @@
 base: &default
   enabled: false
   stub: false
-  write_key: 'null'
+  write_key: true
```
[the remark this answers is not in the document]
Comment (mayorova) [reply config/segment.yml:4]: This is for Segment analytics https://github.com/3scale/porta/blob/99b24fb6622653c0f040e36b5530bd1b55e8f64b/app/views/provider/_analytics.html.erb#L7

We use it in SaaS, on-premises it's disabled. This `true` value was in `openshift` configs, but the value actually doesn't matter, cause as long as it's disabled, it won't be used.

We can actually add env vars here, so we can later reuse this config in SaaS. But not sure whether to leave it for another PR.

File: config/oauth2.yml [hunk 8]
```diff
@@ -14,4 +14,5 @@ test:
     client_secret: fake_secret
 
 production:
-  <<: *default
+  github:
```
[the remark this answers is not in the document]
Comment (mayorova) [reply config/oauth2.yml:17]: This is for "3scale-branded" GitHub integration, that is used in SaaS  *

This has always been disabled in on-premises: https://github.com/3scale/porta/blob/99b24fb6622653c0f040e36b5530bd1b55e8f64b/openshift/system/config/oauth2.yml

* in SaaS we have a global GitHub client, that we allow the providers to use for authentication, without the need to set up their own GitHub OAuth application. They can still use their own GitHub apps, this setting doesn't (or at least shouldn't) prevent that.

File: test/unit/three_scale/middleware/cors_test.rb [hunk 9]
```diff
@@ -76,7 +76,7 @@ class ThreeScale::Middleware::CorsTest < ActiveSupport::TestCase
   end
 
   test 'provider signup path excluded in example configs' do
```
[the remark this answers is not in the document]
Comment (mayorova) [reply test/unit/three_scale/middleware/cors_test.rb]: I called it "default configs" here https://github.com/3scale/porta/pull/3766/commits/cc4a0b56fb883cbf1269e2eefbd8be43614ed606

File: test/unit/three_scale/middleware/cors_test.rb [hunk 10]
```diff
@@ -76,7 +76,7 @@ class ThreeScale::Middleware::CorsTest < ActiveSupport::TestCase
...
-    ["config/examples"].each do |config_dir|
+    ["config"].each do |config_dir|
```
[the remark this answers is not in the document]
Comment (mayorova) [reply test/unit/three_scale/middleware/cors_test.rb]: There was a small extra step for fetching the specific env before using the value for stubbing, it's done in https://github.com/3scale/porta/pull/3766/commits/cc4a0b56fb883cbf1269e2eefbd8be43614ed606

File: config/segment.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (jlledom) [reply config/segment.yml:4]: If it's enabled by default in all environment, how they enable it in SaaS? I guess they need to create a configmap for this file and mount it. I think it would be better to add an env variable to enable it, and leave it disable by default. Anyway, better to do it in another PR I think.

File: config/oauth2.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (jlledom) [reply config/oauth2.yml:17]: Then I think it should be disabled by default. With current configuration it's enabled by for development, but with empty id and key:

```yaml
base: &default
  github:
    enabled: <%= ENV.fetch('GITHUB_ENABLED', '1') == '1' %>
    client_id: <%= ENV['GITHUB_CLIENT_ID'] %>
    client_secret: <%= ENV['GITHUB_CLIENT_SECRET'] %>

development:
  <<: *default
```

IMO it should be like this:
```yaml
    enabled: <%= ENV.fetch('GITHUB_ENABLED', '0') == '1' %>
```

[the remark this answers is not in the document]
Comment (mayorova) [reply config/oauth2.yml:17]: Yep, makes sense :+1:

**Review by jlledom — DISMISSED** [review]

File: config/segment.yml [hunk 11]
```diff
@@ -1,7 +1,6 @@
 base: &default
-  enabled: false
-  stub: false
-  write_key: 'null'
+  enabled: <%= ENV['SEGMENT_WRITE_KEY'].present? %>
```
Comment (akostadinov) [inline config/segment.yml:2]: nice!

**Review by jlledom — DISMISSED** [review]

File: config/database.yml [hunk 12]
```diff
@@ -38,9 +39,10 @@ production:
 <% else; db_host = ENV.fetch('DB_PORT_3306_TCP_ADDR') { ENV['DB_PORT'] ? 'db' : ENV['DB_HOST'] } || '127.0.0.1' %>
 base: &default
   adapter: mysql2
+  url: <%= ENV['DATABASE_URL'] %>
   encoding: utf8
+  pool: <%= ENV.fetch('RAILS_MAX_THREADS', 5) %>
```
Comment (mayorova) [inline config/database.yml:44]: same as https://github.com/3scale/porta/pull/3766/files#r1624383012

File: test/unit/three_scale/middleware/cors_test.rb [hunk 13]
```diff
@@ -75,21 +75,19 @@ class ThreeScale::Middleware::CorsTest < ActiveSupport::TestCase
...
-        middleware = ThreeScale::Middleware::Cors.new(app)
-        status, headers = middleware.call(env.merge('REQUEST_METHOD' => 'OPTIONS', 'HTTP_ACCESS_CONTROL_REQUEST_METHOD' => 'GET', 'PATH_INFO' => provider_signup_path))
-        assert_equal 403, status
-        assert_not headers['Access-Control-Allow-Origin']
-      end
+  test 'provider signup path excluded in default configs' do
+    cors_config = YAML.load_file(Rails.root.join("config/cors.yml")).deep_symbolize_keys
+    rails_envs = %i[development test production]
+    rails_envs.each do |rails_env|
+      stub_config = cors_config[rails_env]
```
Comment (mayorova) [inline test/unit/three_scale/middleware/cors_test.rb:82]: Removed `enabled: true` as it doesn't make difference.

**Review by jlledom — APPROVED** [review]

**Comment by mayorova** [comment]
Production configs in current master (with the config files from `openshift/system/config`):

[openshift-prod.txt](https://github.com/user-attachments/files/15550512/openshift-prod.txt)

Production configs in this branch (no need to copy anything):

[conf-prod.txt](https://github.com/user-attachments/files/15550521/conf-prod.txt)

File: config/zync.yml [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+  skip_non_oidc_applications: false
+
+development:
+  <<: *default
+
+test:
+  <<: *default
+  endpoint:
+
+production:
```
[the remark this answers is not in the document]
Comment (mayorova) [reply config/zync.yml]: > * do we need it? can't we use the built-in `zync.yml` file?

I updated the file in https://github.com/3scale/porta/pull/3766/commits/b8fdb59bf1d99e65283fefdbc386d10a1f34b3e6 a bit, to apply the timeouts that are set in the operator. We can add `ZYNC_ENDPOINT` to the operator and use the default file instead. Possibly the idea of having the `zync.yml` file set in operator is that this way it is available as a ConfigMap, and can be easily changed. On the other hand, I don't think that this file is supposed to be updated easily.

> * if we have one already in `config`, would operator override it?

The extra files defined by the operator are mounted at `/opt/system-extra-configs`, which are then symlinked to `config` directory via https://github.com/3scale/porta/blob/cleanup-config-files-remove-dups/openshift/system/entrypoint.sh#L17

As we use `ln -sf` there shouldn't be a problem with overriding the existing file with the new content (set by operator).

File: .circleci/config.yml [hunk 15]
```diff
@@ -71,11 +71,6 @@ generate-assets-checksums: &generate-assets-checksums
     name: Generate assets precompilation checksum data
     command: git ls-tree -r HEAD $(<.circleci/asset_paths) > tmp/assets_related_checksums
 
-use-example-config-files: &use-example-config-files
-  run:
-    name: Copy example config files into place to be used by tests
-    command: cp config/examples/*.yml config/
```
Comment (akostadinov) [inline .circleci/config.yml:77]: Wait, why do we remove this?

File: .dockerignore
Comment (akostadinov) [inline .dockerignore:1]: I don't understand why are these lines removed.

File: .gitignore
Comment (akostadinov) [inline .gitignore:1]: Same here, these ignores should stay.

File: Dockerfile [hunk 16]
```diff
@@ -19,7 +19,6 @@ ENV DISABLE_SPRING="true" \
 WORKDIR /opt/system/
 
 ADD . ./
-ADD config/examples/*.yml config/
```
Comment (akostadinov) [inline Dockerfile:22]: We need to keep this.

File: bin/setup [hunk 17]
```diff
@@ -20,9 +20,6 @@ FileUtils.chdir APP_ROOT do
   # Install JavaScript dependencies if using Yarn
   # system('bin/yarn')
 
-  puts "\n== Copying sample files =="
-  cp_r 'config/examples/.', 'config/', preserve: true, dereference_root: true
```
Comment (akostadinov) [inline bin/setup:24]: I think we should not overwrite files, is there such an option?

File: config/amazon_s3.yml
Comment (akostadinov) [inline config/amazon_s3.yml:1]: These files contain secrets. The should be git ignored and dockerignored. To prevent leaking them into git or into the container image builds.

**Comment by akostadinov** [comment]
I think we shouldn't change the structure of `config/examples`. These should be the files we do local modifications in our local dev envs. Such that when we do local modifications, these don't show up in `git diff` and don't leak into github or a locally built container image.

File: config/database.yml (hunk 12 above)
Comment (jlledom) [reply config/database.yml:44]: Shouldn't this be `utf8mb3` or `utf8mb3_bin` like in `schema.rb`? Just asking, I don't really know the difference between encoding, charset and collation.

https://github.com/3scale/porta/blob/43dcd03310c7d6534fabaf3c85af81b64d9a0146/db/schema.rb#L15

File: .circleci/config.yml (hunk 15 above)
Comment (jlledom) [reply .circleci/config.yml:77]: Because there's no `config/examples` anymore, this PR removes it and moves its files to `config.`

File: Dockerfile (hunk 16 above)
Comment (jlledom) [reply Dockerfile:22]: No, `config/examples` doesn't exist anymore and its files are already under `config`

File: .dockerignore
Comment (jlledom) [reply .dockerignore:1]: Before, the Dockerfile copied them from `config/examples`, so the ones under `config` had to be ignored, then overwritten by those under `config/examples`.

Now they have been moved from `config/examples` so the ones under `config` are already the good ones and we can keep them.

File: .gitignore
Comment (jlledom) [reply .gitignore:1]: No, now we want to track them in git as we tracked those under `config/examples` before, since they are essentially the same files.

File: config/amazon_s3.yml
Comment (jlledom) [reply config/amazon_s3.yml:1]: This one doesn't contain secrets:

https://github.com/3scale/porta/blob/cleanup-config-files-remove-dups/config/amazon_s3.yml

And it was already in git before, but under `config/examples`

**Comment by jlledom** [comment]
> I think we shouldn't change the structure of `config/examples`. These should be the files we do local modifications in our local dev envs. Such that when we do local modifications, these don't show up in `git diff` and don't leak into github or a locally built container image.

I think I expressed this concern before but I don't find the comment. Anyway, this PRs aims to leave the files under `config` immutable and use ENV variables to set our local configs from the `.env` file.

**Comment by akostadinov** [comment]
It's not so good idea to keep passwords in env variables because they are visible to all local processes from this or any other user.

**Comment by mayorova** [comment]
> I think we shouldn't change the structure of `config/examples`. These should be the files we do local modifications in our local dev envs. Such that when we do local modifications, these don't show up in `git diff` and don't leak into github or a locally built container image.

> I think I expressed this concern before but I don't find the comment. Anyway, this PRs aims to leave the files under `config` immutable and use ENV variables to set our local configs from the `.env` file.

@akostadinov @jlledom Yes, I understand your concerns. But what @jlledom says is exactly the reasoning.

I actually found it pretty annoying many times that when some PR changes some configs, you need to copy the `.yml` files from examples to configs again and again, because they are not "in sync" with the branch.

But I also agree that sometimes we want to test stuff locally, and we don't want necessarily to affect the "default" configs.

My idea was to be able to configure **most** of the things (all would be impossible) via environment variables, and `.env` always stays local. Now, to figure out which settings are commonly modified for local development might take some time, and we may want to add env vars gradually.

But if you don't like this approach, I'm fine with just moving all the configs back to `config/examples` again, no problem. In the Dockerfile builds we'll just copy them from `config/examples` rather than from `openshift/system/config`.

**Comment by akostadinov** [comment]
At least for anything with passwords, need to be in examples. The rest I guess makes more sense to stay as defaults.

**Comment by jlledom** [comment]
> It's not so good idea to keep passwords in env variables because they are visible to all local processes from this or any other user.

I don't thinks that's much of a problem. I have a few passwords stored in my .env and all of them are for fake/test/development/staging environments, I never needed to use any production credentials.

**Comment by akostadinov** [comment]
Any cloud credentials can equally well be abused and costs incurred.

**Comment by mayorova** [comment]
Well, regarding the passwords, in the image the passwords would need to go in env vars anyway, so if we want to use the default configs for both local development and images, they will need to be in env vars in any case.

I think if these are stored in `.env` in the project dir, they would be read by `dotenv` only when the Rails process is started, I don't think they are shared with other processes in this case.

I am more concerned about such files as `settings.yml` where we sometimes tweak `onpremises: true`, or `force_ssl:` and `secure_cookie:` (which are currently hardcoded to `true` for production, which is good for openshift, but not good for local env - here for example we might bring back `FORCE_SSL` env var with a default value).

Or maybe `domain_substitution.yml` which sometimes is tweaked when we run against the staging DB.

Or potentially `rolling_updates.yml`, however I don't think we ever touch it nowadays, and always just apply the defaults.

Probably in `core.yml` we could add an env var for `fake_server`.

So, we can maybe put `settings.yml`, `core.yml` and `domain_substitution.yml` back to `examples`, or we can add env vars for the values that will potentially change.

I'd vote for env vars way. I think it's easier, and we can even just have multiple sets of `.env` files with different setups (e.g. "production mode pointing to staging", "local saas" etc.), while config files always stay the same.

**Comment by akostadinov** [comment]
I see your concerns. Just an idea, this might make sense to be implemented with a global override configuration file. Where one can put any configration value and it will override the defaults. Just for dev and test environment. It would not be trivial to implement but would probably cover all use cases.

> Well, regarding the passwords, in the image the passwords would need to go in env vars anyway, so if we want to use the default configs for both local development and images, they will need to be in env vars in any case.

It is alright to have env vars read by the config files. Just we locally change it to a static values for sensitive content. 

> I think if these are stored in .env in the project dir, they would be read by dotenv only when the Rails process is started, I don't think they are shared with other processes in this case.

Check `/proc/<PID>/environ`, e.g. `/proc/406847/environ`, you can read the environment of any running process by any other running process on your system.

File: .circleci/config.yml (hunk 15 above)
Comment (mayorova) [reply .circleci/config.yml:77]: @akostadinov @jlledom if you prefer, I can move them back to `config/examples`, so the only thing the PR will do is get rid of the duplicates in `openshift/system/config` by  unifying the files

Comment (akostadinov) [reply .circleci/config.yml:77]: I prefer returning the examples until we have something to deal with setting credentials in a relatively secure way.

File: bin/setup (hunk 17 above)
Comment (akostadinov) [reply bin/setup:24]: nothing, mistake

File: config/amazon_s3.yml
Comment (akostadinov) [reply config/amazon_s3.yml:1]: it contains `secret_access_key`, at least locally is supposed to

**Comment by akostadinov** [comment]
> three_scale.segment.write_key

Why should it be `true`? `nil` makes more sense for something that is not used.

**Comment by mayorova** [comment]
> > three_scale.segment.write_key
> 
> Why should it be `true`? `nil` makes more sense for something that is not used.

Yeah, sorry, that comment is outdated, since then I update to taking the value from an env var: https://github.com/3scale/porta/pull/3766/files#diff-3027cb92d1f25cc230471a202a313a8e4ffdc445147f6530c29dbc6e7b7e3ef8R3

So, if the env var is not set, it will be `nil`. `true` was confusing indeed, the only reason I even used this value because it was like that in the current `openshift` config: https://github.com/3scale/porta/blob/505d6ccfb919d2a20eeb9cf7474468b94a149c60/openshift/system/config/segment.yml#L4

**Comment by mayorova** [comment]
@akostadinov @jlledom I restored https://github.com/3scale/porta/pull/3766/commits/b9dd0224590666c08d39ac7fe3ad6b124bb5a446

You can compare the contents of the config files in `production` mode (so, what will be applied in the image - note that the operator will overwrite some files):
1. master, with `openshift/system/config` files copied to `config`
[openshift-master.txt](https://github.com/user-attachments/files/15738649/openshift-master.txt)

2. this branch, with both `openshift/system/config` and `config/examples` files copied to `config`
[openshift-cleanup.txt](https://github.com/user-attachments/files/15738650/openshift-cleanup.txt)

**Review by josemigallas — APPROVED** [review]

**Review by jlledom — APPROVED** [review]

File: bin/setup (hunk 17 above)
Comment (jlledom) [reply bin/setup]: Well, after https://github.com/3scale/porta/pull/3766/commits/b9dd0224590666c08d39ac7fe3ad6b124bb5a446 this change has been reverted so we are overwriting again @akostadinov

**Comment by mayorova** [comment]
@akostadinov what do you think now?

File: openshift/system/Dockerfile [hunk 18]
```diff
@@ -6,6 +6,7 @@ ENV RUBY_MAJOR_VERSION=2 \
     NODE_ENV=production
 ENV RUBY_VERSION="${RUBY_MAJOR_VERSION}.${RUBY_MINOR_VERSION}"
 ADD . ./
+ADD config/examples/*.yml ./config/
```
Comment (akostadinov) [inline openshift/system/Dockerfile:9]: But these files are already in the image, I wonder if this is better or it is better to later do `cp config/examples/* config/`

But should be good either way.

Pls don't forget to backport this change in the production Dockerfile also.

**Review by akostadinov — APPROVED** [review]
Didn't compare all configs, looked at the rest of the stuff. Looks nice! Seems like a lot of good work!

File: openshift/system/Dockerfile (hunk 18 above)
Comment (mayorova) [reply openshift/system/Dockerfile:9]: Yeah, we could do that too. Let me take a look.

Comment (mayorova) [reply openshift/system/Dockerfile]: Like this @akostadinov ?
https://github.com/3scale/porta/pull/3766/commits/d941b5afe0ec81eabfefa96876df47a541f868ad

Comment (jlledom) [reply openshift/system/Dockerfile]: Why is `cp` better than `ADD`?

Comment (mayorova) [reply openshift/system/Dockerfile]: I am not sure :grimacing: But in this case I think the benefit is that we create just one intermediate image instead of two.

Comment (jlledom) [reply openshift/system/Dockerfile]: ok

**Review by jlledom — APPROVED** [review]

**Review by akostadinov — APPROVED** [review]
